buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -538,30 +538,38 @@ static void setBytes(AbstractByteBuf buf, long addr, int index, byte[] src, int
     }
 
     static void setBytes(AbstractByteBuf buf, long addr, int index, ByteBuffer src) {
-        buf.checkIndex(index, src.remaining());
```

Comment:
@franz1981 I think this should always come first.

Reply:
I was thinking the same but it could be performed twice, because internalNio etc etc is doing the same

Reply:
Add a comment please then.

Reply:
@normanmaurer Added on `setSingleBytes`